src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -570,6 +570,39 @@ void Simulator::poll_topics()
 	if (updated) {
 		orb_copy(ORB_ID(vehicle_status), _vehicle_status_sub, &_vehicle_status);
 	}
+
+	// send tune control message if available
+
+	orb_check(_tune_control_sub, &updated);
+
+	if (updated) {
+
+		orb_copy(ORB_ID(_control), _control_sub, &_tune_control);
+
+		_tunes.set_control(_tune_control);
+
+		mavlink_play_tune_t tune_msg = { };
+		mavlink_message_t message = { };
+
+		struct {
+			unsigned int frequency;
+			unsigned int duration;
+			unsigned int silence;
+		} tune_info = {};
+
+		_tunes.get_next_note(tune_info.frequency, tune_info.duration, tune_info.silence);
```

Comment:
I think I raised this before, you need to move this out of the `updated` condition. A single tune_control topic update might lead to several notes.